app/src/main/java/com/jerboa/db/AppDB.kt

```diff
@@ -224,10 +256,26 @@ class AccountRepository(private val accountDao: AccountDao) {
     }
 }
 
+class SearchHistoryRepository(
+    private val searchHistoryDao: SearchHistoryDao,
+    private val appSettingsDao: AppSettingsDao,
+) {
+    fun history(): Flow<List<SearchHistory>> = searchHistoryDao.history()
+
+    suspend fun insert(item: SearchHistory) {
+        if (appSettingsDao.settings().first().saveSearchHistory) {
```

Comment:
Why is this necessary? It should be connected to an account, not the appsettings, so it'll already have the account.id in the form.

There's also no need to do a check on this insert. If you have an appsetting to disable search, that should be done outside of / before this SQL logic.